### PR TITLE
Add script to clean repo data with GitHub API

### PR DESCRIPTION
**Pull request by konstin**
This adds a script that queries the GitHub repo API to prune known-github-tomls.jsonl into known-github-tomls-clean.jsonl.

This solves three problems that i ran into when trying to scale the ecosystem CI with known-github-tomls.jsonl:

1. The original known-github-tomls.jsonl contains forks, which inflate some data incorrectly. There, we check in the API response whether the repo is a fork and exclude those.
2. Sometimes the `ref` is wrong, which would make my clones fail. We check the GitHub API for the default branch and use those.
3. Some entries in known-github-tomls.jsonl do not contain a `path`. Those are filtered out.

To avoid running into GitHub API rate limits, the results are cached into repo_api_data.jsonl

### CODE REVIEW

File: ruff_usage_aggregate/actions/clean_with_repo_api.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+):
+    repos = list(read_jsonl(known_github_tomls))
+    paths = {}
+    for repo in repos:
+        if path := repo.get("path"):
+            paths[(repo["owner"], repo["repo"])] = path
+    known_info: dict[(str, str), RepoInfo] = {}
+    for entry in read_jsonl(repo_api_data):
+        if path := paths[(entry["owner"], entry["repo"])]:
+            known_info[(entry["owner"], entry["repo"])] = RepoInfo(path=path, **entry)
```
Comment (konstin) [inline ruff_usage_aggregate/actions/clean_with_repo_api.py:89]: this is a bit awkward because the path is the only thing we don't get from the github api

**Review by akx — APPROVED** [review]
Cool, thanks!

**Review by akx — COMMENTED** [review]
Lint's failing – care to run `pre-commit run --all-files`?

**Comment by konstin** [comment]
sorry missed the pre-commit, should be fixed now

**Comment by akx** [comment]
All good, thanks @konstin!

**Pull request merged**